src/NewChain/AuxiliaryChain.ts

```diff
@@ -274,12 +275,80 @@ export default class AuxiliaryChain {
     node.startSealer(gasPrice, targetGasLimit, bootKeyFile);
     // The sealer runs locally on this machine and the port is published to the host from the
     // docker container.
-    this.logInfo('waiting 5 seconds for the sealer port to become available');
-    await AuxiliaryChain.sleep(5000);
     // Has to be RPC and not WS. WS connection was closed before deploying the Co-Gateway.
     // Reason unknown. Possibly due to the fact that according to `lsof` node keeps opening new
     // connections.
     this.web3 = new Web3(`http://127.0.0.1:${this.nodeDescription.rpcPort}`);
+    await this.verifyAccountsUnlocking();
+  }
+
+  /**
+   * It polls every 4-secs to fetch the list of wallets.
+   * It logs error when connection is not established even after max tries.
+   */
+  private async verifyAccountsUnlocking() {
+    let noOfRetries;
+    let unlockStatus;
+    for (noOfRetries = 1; noOfRetries <= this.maxRetries; noOfRetries++) {
+      this.logInfo(`number of tries to fetch unlocked accounts is ${noOfRetries}`);
+      let response;
+      await AuxiliaryChain.sleep(4000);
+      try {
+        response = await this.getUnlockedAccounts();
+        if (response.result.length > 0) {
+          unlockStatus = this.getAccountsStatus(response.result);
+          if (unlockStatus) {
+            this.logInfo('accounts unlocked');
+            break;
+          }
+        }
+      }
+      catch (err) {
+        this.logError(err);
+      }
+    }
+    if(!unlockStatus) {
+      this.logError('accounts not unlocked');
```

Comment:
Shouldn't it `throw` instead? The chain creation cannot continue without unlocked accounts. Either throw here, or throw at the caller.